src/library_syscall.js

```diff
@@ -603,8 +603,17 @@ var SyscallsLibrary = {
       if (stream.stream_ops.poll) {
         var timeoutInMillis = -1;
         if (timeout) {
-          var tv_sec = (readfds ? {{{ makeGetValue('timeout', C_STRUCTS.timeval.tv_sec, 'i32') }}} : 0),
-              tv_usec = (readfds ? {{{ makeGetValue('timeout', C_STRUCTS.timeval.tv_usec, 'i32') }}} : 0);
+          // select(2) is declared to accept "struct timeval { time_t tv_sec; suseconds_t tv_usec; }".
+          // However, musl passes the two values to the syscall as an array of long values.
+          // Note that sizeof(time_t) != sizeof(long) in wasm32. The former is 8, while the latter is 4.
+          // This means using "C_STRUCTS.timeval.tv_usec" leads to a wrong offset.
+          // So, we use an immediate offset 4 for wasm32, and 8 for wasm64.
+          var tv_sec = (readfds ? {{{ makeGetValue('timeout', 0, 'i32') }}} : 0),
+#if MEMORY64
+              tv_usec = (readfds ? {{{ makeGetValue('timeout', 8, 'i32') }}} : 0);
+#else
+              tv_usec = (readfds ? {{{ makeGetValue('timeout', 4, 'i32') }}} : 0);
+#endif
```

Comment:
Thanks!  I had no idea musl did this.  

BTW you can just POINTER_SIZE here instead of 4/8.

Reply:
Thanks, I tweaked my PR by POINTER_SIZE. I thought there must be something like that, but I couldn't find it.

I am also fine to patch musl if you prefer it. But it might be a little tedious hunk when updating musl.